### ISSUE TITLE
Add a no-JS Admin page

### ISSUE THREAD

**Issue by davwheat** [flarum/core#2911]
## Feature Request

**Is your feature request related to a problem? Please describe.**
If there is a backend/boot/JS error, we should still allow admins to access the admin page.

This could be done with a simple No-JS admin page. This could be something like `localhost/admin?safe=true`. If this `safe=true` is present, we should **not** load any extensions while displaying this page. We can then show the user a page that allows them to make changes to the active extensions (enabling/disabling).

**Describe the solution you'd like**
TBD

**Justify why this feature belongs in Flarum's core, rather than in a third-party extension**
This is a core part of Flarum that would improve accessibility to more users.

**Describe alternatives you've considered**
TBD

**Comment by askvortsov1** [comment]
I've been thinking more about the safe more option, and I'm leaning towards going with something similar to Discourse's approach. I don't think we should allow disabling extensions on the backend via safe mode because that opens up a whole slew of potential security issues. Furthermore, it couples per-request state to global application state, which is something we'd like to avoid.

I started working on a safe mode PR, but stopped when I realized I didn't understand the use case. What do we want this to do? Do we want to make it easier to debug frontend issues? Offer a way for admins to rescue a "bricked" forum by disabling all extensions? Allow admins to edit extension settings in the admin panel without those reflecting on the forum side?

**Comment by SychO9** [comment]
> I'm leaning towards going with something similar to Discourse's approach.

How does discourse handle it ?

> Do we want to make it easier to debug frontend issues? Offer a way for admins to rescue a "bricked" forum by disabling all extensions? Allow admins to edit extension settings in the admin panel without those reflecting on the forum side?

No, yes, no.

> What do we want this to do?

If we take the admin NoJS view as an example, which solves the issue of an extension breaking the admin frontend and locking out the admin from fixing it by disabling said extension. The same can happen if an extension breaks the backend and prevents Flarum from booting, the admin NoJS view is only solving half of the problem, having the possibility to safely boot to disable said extensions and therefor require less SSH access is needed.

I don't know what exactly you're referring to when mentioning security issues, but if my assumptions are correct, I think the smart thing to do is to have safe mode as sort of an offline mode level 2, where it is an offline mode (which locks users from accessing the forum apart from the admin) but it also doesn't boot any extensions.

> Furthermore, it couples per-request state to global application state, which is something we'd like to avoid.

Isn't that why we're avoiding the URL safe mode option, or is it also an issue with other approaches ?

**Comment by askvortsov1** [comment]
Makes sense, I think I was overthinking this when I wrote it. I'll open a PR for this.

> How does discourse handle it ?

Discourse's approach is that any user can trigger a ["no-js safe mode"](https://meta.discourse.org/t/how-to-use-discourse-safe-mode/53504) where the backend works with all plugins, but plugin JS isn't booted. Apparently, the goal is to help diagnose broken JS issues. 

I believe that when we convert our JS system to be more extender based, a lot more errors will happen on boot, meaning that diagnosing where stuff goes wrong will become a lot easier. I think that might be a more effective system than just disabling JS.
